sdk/v2/core/projects.go

```diff
@@ -0,0 +1,327 @@
+	// must be labeled. An event must have ALL of these labels to match this
+	// subscription.
+	Labels Labels `json:"labels,omitempty"`
+}
+
+// KubernetesConfig represents Kubernetes-specific configuration. This is used
+// primarily at the Project level, but is also denormalized onto Events so that
+// Event handling doesn't required a Project lookup to obtain
+// Kubernetes-specific configuration.
+type KubernetesConfig struct {
+	// Namespace is the dedicated Kubernetes namespace for the Project. This is
+	// NOT specified by clients when creating a new Project. The namespace is
+	// created by / assigned by the system. This detail is a necessity to prevent
+	// clients from naming existing namespaces in an attempt to hijack them.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// ProjectsClient is the specialized client for managing Projects with the
+// Brigade API.
+type ProjectsClient interface {
+	// Create creates a new Project.
+	Create(context.Context, Project) (Project, error)
+	// CreateFromBytes creates a new Project using raw (unprocessed by the client)
+	// bytes, presumably originating from a file. This is the preferred way to
+	// create Projects defined by an end user since server-side validation will
+	// then be applied directly to the Project definition as the user has written
+	// it (i.e. WITHOUT any normalization or corrections the client may have made
+	// when unmarshaling the original data or when marshaling the outbound
+	// request).
+	CreateFromBytes(context.Context, []byte) (Project, error)
+	// List returns a ProjectList, with its Items (Projects) ordered
+	// alphabetically by Project ID.
+	List(context.Context, ProjectsSelector, meta.ListOptions) (ProjectList, error)
+	// Get retrieves a single Project specified by its identifier.
+	Get(context.Context, string) (Project, error)
+	// Update updates an existing Project.
+	Update(context.Context, Project) (Project, error)
+	// UpdateFromBytes updates an existing Project using raw (unprocessed by the
+	// client) bytes, presumably originating from a file. This is the preferred
+	// way to update Projects defined by an end user since server-side validation
+	// will then be applied directly to the Project definition as the user has
+	// written it (i.e. WITHOUT any normalization or corrections the client may
+	// have made when unmarshaling the original data or when marshaling the
+	// outbound request).
+	UpdateFromBytes(context.Context, string, []byte) (Project, error)
+	// Delete deletes a single Project specified by its identifier.
+	Delete(context.Context, string) error
+
+	// Roles returns a specialized client for Project Role management.
+	Roles() ProjectRolesClient
+
+	// Secrets returns a specialized client for Secret management.
+	Secrets() SecretsClient
+}
+
+type projectsClient struct {
+	*restmachinery.BaseClient
+	// rolesClient is a specialized client for Project Role managament.
```

Comment:
typo: _managament_

Reply:
Ok... _now_ I'm back to wondering if the spellcheck linter was on a coffee break.